tests/unittests/tests-gnrc_ipv6_nib/tests-gnrc_ipv6_nib-internal.c

```diff
@@ -964,6 +965,735 @@ static void test_nib_drl_get_dr__success4(void)
     TEST_ASSERT(nib_res != node2);
 }
 
+#if GNRC_IPV6_NIB_NUMOF < GNRC_IPV6_NIB_OFFL_NUMOF
+#define MAX_NUMOF   (GNRC_IPV6_NIB_NUMOF)
+#else /* GNRC_IPV6_NIB_NUMOF < GNRC_IPV6_NIB_OFFL_NUMOF */
+#define MAX_NUMOF   (GNRC_IPV6_NIB_OFFL_NUMOF)
+#endif
+
+/*
+ * Creates MAX_NUMOF off-link entries with different next-hop addresses and
+ * then tries to add another.
+ * Expected result: should return NULL
+ */
+static void test_nib_offl_alloc__no_space_left_diff_next_hop(void)
+{
+    ipv6_addr_t next_hop = { .u64 = { { .u8 = LINK_LOCAL_PREFIX },
+                                    { .u64 = TEST_UINT64 } } };
+    static const ipv6_addr_t pfx = { .u64 = { { .u8 = GLOBAL_PREFIX } } };
+
+    for (int i = 0; i < MAX_NUMOF; i++) {
+        _nib_offl_entry_t *dst;
+
+        TEST_ASSERT_NOT_NULL((dst = _nib_offl_alloc(&next_hop, IFACE, &pfx,
+                                                    GLOBAL_PREFIX_LEN)));
+        TEST_ASSERT_NOT_NULL(dst->next_hop);
+        dst->mode |= _FT;
+        next_hop.u64[1].u64++;
+    }
+    TEST_ASSERT_NULL(_nib_offl_alloc(&next_hop, IFACE, &pfx, GLOBAL_PREFIX_LEN));
+}
+
+/*
+ * Creates MAX_NUMOF off-link entries with different interfaces and then tries
+ * to add another.
+ * Expected result: should return NULL
+ */
+static void test_nib_offl_alloc__no_space_left_diff_iface(void)
+{
+    static const ipv6_addr_t next_hop = { .u64 = { { .u8 = LINK_LOCAL_PREFIX },
+                                                 { .u64 = TEST_UINT64 } } };
+    static const ipv6_addr_t pfx = { .u64 = { { .u8 = GLOBAL_PREFIX } } };
+    unsigned iface = IFACE;
+
+    for (int i = 0; i < MAX_NUMOF; i++) {
+        _nib_offl_entry_t *dst;
+        TEST_ASSERT_NOT_NULL((dst = _nib_offl_alloc(&next_hop, iface, &pfx,
+                                                    GLOBAL_PREFIX_LEN)));
+        TEST_ASSERT_NOT_NULL(dst->next_hop);
+        dst->mode |= _PL;
+        iface++;
+    }
+    TEST_ASSERT_NULL(_nib_offl_alloc(&next_hop, iface, &pfx, GLOBAL_PREFIX_LEN));
+}
+
+/*
+ * Creates MAX_NUMOF off-link entries with different next-hop addresses and
+ * interfaces and then tries to add another.
+ * Expected result: should return NULL
+ */
+static void test_nib_offl_alloc__no_space_left_diff_next_hop_iface(void)
```

Comment:
I dont' think you need the above two test cases when you introduce this test case that combines both